android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -58,27 +72,48 @@ public void finishNativeInitialization() {
         initState();
     }
 
+    @SuppressLint("NotifyDataSetChanged")
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode != RESULT_OK) return;
+
+        BraveActivity activity = BraveActivity.getBraveActivity();
+        if (activity != null) {
+            activity.getWalletModel().getCryptoModel().getNetworkModel().init();
+        }
+        networkSelectorAdapter.notifyDataSetChanged();
```

Comment:
shouldn't this be handled via the live data changes? do we need an explicit notify call?

Reply:
> shouldn't this be handled via the live data changes?

Correct! Not needed anymore, I'm removing it.